code/datums/atom_hud.dm

```diff
@@ -56,8 +56,10 @@ GLOBAL_LIST_INIT(huds, list(
 
 
 /datum/atom_hud/proc/remove_from_hud(atom/A)
+	SIGNAL_HANDLER
 	if(!A)
 		return FALSE
+	UnregisterSignal(A, COMSIG_PARENT_QDELETING)
```

Comment:
is it better to istype before unregistering? I dunno

Reply:
why would it

Reply:
Do images have signal tables